hw/top_earlgrey/data/top_earlgrey.hjson

```diff
@@ -475,6 +478,13 @@
       clock_srcs: {clk_i: "main", clk_edn_i: "main"},
       clock_group: "trans",
       reset_connections: {rst_ni: "sys", rst_edn_ni: "sys"},
+      param_decl: {
+        Masking: "1",
+        SBoxImpl: "aes_pkg::SBoxImplDom",
+        SecStartTriggerDelay: "0",
+        SecAllowForcingMasks: "1'b0",
```

Comment:
These two parameters are security critical. We must make sure that for the ASIC, we don't overwrite such parameters with non-default values. If I understand it correctly, we still need to list them here such that topgen can expose them e.g. for the FPGA chiplevel in the first place. So if this template was modified, it could still happen that we set non-default values.

I think in the past topgen was able to automatically infer module parameters and default values from the module hjson. But there are probably reasons for doing it this way. Anyway, it doesn't make a difference. We will have to check that `top_earlgrey.sv` doesn't provide non-default values for these parameters before tapeout.

Reply:
ah yes, so i wanted to simplify the auditing process. By this mean at the `chip_earlgry_asic` level, we remove all (except for 1 pad related) parameters.  This means we don't have to double check at that level we did anything incorrect.

For the secure parameters you're talking about, i can also remove them from the instance here.  I was honestly just copying over everything set at the chip_earlgrey level into the hjson. 

Let me know what would work best.  The hope here is to limit the number of mistakes for the taped out version and have one common place we can review.

Reply:
i'm going to merge this in first, but please let me know, and we can adjust as needed.

Reply:
Thanks @tjaychen for the explanation, this is fine.

Reply:
@vogelpi just to clarify one last thing. The parameter declaration is mostly for us to declare a different parameter level at the hjson when we DON'T want the module default. 

So if i take it out from here, the default assumed in Aes would directly apply.  Should i do that instead? It sounds like what you're saying is that the `SecParameters` we really want them to default to the secure version in the ip hjson.  The ASIC version should never set it, and we should only see it touched for FPGA / verilator, is that right?

Reply:
Yes, that is 100% correct Tim. What I am not sure about (it has been a while since I last looked into this): if we take the parameters out from here, can the FPGA chiplevel still change it?

Reply:
yes it can.  If we don't set it here, it just means for the asic version we do not intend to override the value.
The parameter will still be propagated to the top level using the Aes default. 

Let me do that quickly.

Reply:
Oh, great. That is the perfect solution then! Thanks for the new PR, it looks good.